app/backend/approaches/retrievethenread.py

```diff
@@ -70,12 +71,22 @@ def run(self, q: str, overrides: dict[str, Any]) -> Any:
         content = "\n".join(results)
 
         prompt = (overrides.get("prompt_template") or self.template).format(q=q, retrieved=content)
-        completion = openai.Completion.create(
-            engine=self.openai_deployment, 
-            prompt=prompt, 
-            temperature=overrides.get("temperature") or 0.3, 
-            max_tokens=1024, 
-            n=1, 
-            stop=["\n"])
+
+        if(self.openai_type == "azure"):
+            completion = openai.Completion.create(
+                engine=self.openai_deployment, 
+                prompt=prompt, 
+                temperature=overrides.get("temperature") or 0.3, 
+                max_tokens=1024, 
+                n=1, 
+                stop=["\n"])
+        else:
```

Comment:
These two conditions look almost identical? Is there a way to reduce duplication in this code